esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -172,12 +173,12 @@ def main(cfg):
     weights_path = Path(input_files[0]) / filename
     weights = read_weights(weights_path)
 
-    models = read_metadata(cfg)['tas']
+    models = read_metadata(cfg)['tas_CLIM_future']
     model_data, model_data_files = read_model_data(models)
 
     # if a historical period is given calculate the change
-    if 'tas_reference' in read_metadata(cfg).keys():
-        models_hist = read_metadata(cfg)['tas_reference']
+    if 'tas_CLIM_reference' in read_metadata(cfg).keys():
+        models_hist = read_metadata(cfg)['tas_CLIM_reference']
```

Comment:
No need to read metadata 3 times here ;-)

```suggestion
    if 'tas_CLIM_reference' in metadata:
        models_hist = metadata['tas_CLIM_reference']
```